### PR TITLE
simplify logic for func BuildArgumentListFromMap

### PR DESCRIPTION
**Pull request by jiaxuanzhou**
**What this PR does / why we need it**:
simplify the logic to merge two map, we just need only one sort to build the command and make the code more readable.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jiaxuanzhou** [comment]
@dixudx please help review

**Review by rosti — CHANGES_REQUESTED** [review]
Hi @jiaxuanzhou . Thanks for your contribution!
This looks way better than before, however it is not an expected behavior for the caller of `BuildArgumentListFromMap` to have `baseArguments` modified by it. Therefore, we might actually need another for loop to start with a clean copy of `baseArguments` and use that instead.

**Review by kad — CHANGES_REQUESTED** [review]
@rosti is correct. baseArguments parameter is not a copy, but pointer. Modifying it inside BuildArgumentListFromMap function, means modification to original map. So, simplification is possible, but first baseArguments must be copied to new variable.

**Comment by jiaxuanzhou** [comment]
@rosti @neolit123 please help review again, thx

**Review by neolit123 — COMMENTED** [review]
@jiaxuanzhou LGTM on a quick look.
will defer to others.

please squash the commits into one.

also have a look at the CI errors and try running:
./hack/update-gofmt.sh
./hack/update-bazel.sh

**Comment by jiaxuanzhou** [comment]
@neolit123 i rebased the commits to one, and run the fmt & bazel scripts.

**Comment by neolit123** [comment]
@jiaxuanzhou 
the gofmt and go lint test fails with:
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/67360/pull-kubernetes-verify/111539/

try resolving the issues locally:
./hack/verify-gofmt.sh
./hack/verify-golint.sh

make sure you have go 1.11.x installed.

**Comment by jiaxuanzhou** [comment]
here is error from hack/verify-golint.sh:
```
+ for err in '"${errors[@]}"'
+ echo 'pkg/volume/util/resize_util.go:30:24: should omit type map[v1.PersistentVolumeClaimConditionType]bool from declaration of var knownResizeConditions; it will be inferred from the right-hand side'
pkg/volume/util/resize_util.go:30:24: should omit type map[v1.PersistentVolumeClaimConditionType]bool from declaration of var knownResizeConditions; it will be inferred from the right-hand side
+ echo

+ echo 'Please review the above warnings. You can test via "golint" and commit the result.'
Please review the above warnings. You can test via "golint" and commit the result.
+ echo 'If the above warnings do not make sense, you can exempt this package from golint'
If the above warnings do not make sense, you can exempt this package from golint
+ echo 'checking by adding it to hack/.golint_failures (if your reviewer is okay with it).'
checking by adding it to hack/.golint_failures (if your reviewer is okay with it).
+ echo

+ false
++ kube::log::errexit
++ local err=1
++ set +o
++ grep -qe '-o errexit'
++ set +o xtrace
!!! Error in hack/verify-golint.sh:114
  Error in hack/verify-golint.sh:114. 'false' exited with status 1
Call stack:
  1: hack/verify-golint.sh:114 main(...)
Exiting with status 1
```

i am looking into the errs from test

File: cmd/kubeadm/app/util/arguments.go [hunk 1]
```diff
@@ -30,26 +30,22 @@ import (
...
-	}
-	sort.Strings(keys)
-	for _, k := range keys {
-		v := overrideArguments[k]
-		// values of "" are allowed as well
-		command = append(command, fmt.Sprintf("--%s=%s", k, v))
+
+	for k, v := range baseArguments {
+		if _, overrideExists := overrideArguments[k]; !overrideExists {
+			overrideArguments[k] = v
```
Comment (kad) [inline cmd/kubeadm/app/util/arguments.go]: this still has the issue that was reported before: content of arguments in outer scope of that function will be changed if we do assignment to `baseArguments` or `overrideArguments`

Comment (rosti) [reply cmd/kubeadm/app/util/arguments.go]: @jiaxuanzhou don't modify any of the arguments. Just copy `baseArguments` into another `map[string]string` and then override that copy with the contents of `overrideArguments`. Finally, fetch its keys, sort them and construct the command line options slice to be returned.

Comment (jiaxuanzhou) [reply cmd/kubeadm/app/util/arguments.go]: thx, i got you and modified.

**Comment by jiaxuanzhou** [comment]
@neolit123  please help approve the pr. thx

**Review by timothysc — APPROVED** [review]
/approve 

But I don't think this meets the criteria for slush, so punting till next cycle.

**Review by kad — APPROVED** [review]
/lgtm

**Pull request merged**